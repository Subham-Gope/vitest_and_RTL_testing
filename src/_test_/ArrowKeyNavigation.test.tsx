// import { describe, test, expect } from "vitest";
// import { screen, render } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ArrowKeyNavigation from "../components/ArrowKeyNavigation";

// describe("Testing the ArrowKeyNavigation.tsx", () => {
//   test("clears the input on Escape key press", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     const { container } = render(
//       <ArrowKeyNavigation message={"Hello"} num={5} />
//     );
//     expect(container).toMatchSnapshot();

//     const input = screen.getByLabelText(/enter text/i);
//     expect(input).toBeInTheDocument();

//     // Initiating the user interaction
//     await user.type(input, "Hello");

//     expect(input).toHaveValue("Hello");

//     await user.keyboard("{Escape}");

//     expect(input).toHaveValue("");
//   });
// });
