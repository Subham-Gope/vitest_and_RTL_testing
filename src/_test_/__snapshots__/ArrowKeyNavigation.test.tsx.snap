// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ArrowKeyNavigation.tsx > clears the input on Escape key press 1`] = `
<div>
  <h1>
    Message: 
    Hello
    , Numbers: 
    5
  </h1>
  <div>
    <label
      for="arrow"
    >
      Enter Text
    </label>
    <input
      id="arrow"
      placeholder="enter text"
      type="text"
      value=""
    />
  </div>
</div>
`;
